tdp_core/security/manager/jwt_security_manager.py

```diff
@@ -212,69 +106,117 @@ def logout(self):
-            load_from_req = getattr(store, "load_from_request", None)
-            if callable(load_from_req):
-                user = load_from_req(request)
-                if user:
-                    return user
-
-        # finally, return None if all methods did not login the user
-        return None
+def login_required(f=None, *, users=(), roles=()):
+    """ Usage: @login_required or @login_required(users=("admin") or @login_required(roles=("admin")) """
+    def login_required_inner(fn=None):
+        @wraps(fn)
+        def decorator(*args, **kwargs):
+            u = phoveaSecurityManager().current_user
+            # Allow access only if a user is available
+            if not u:
+                return abort(401, {'message': 'No user in login_required request'})
+            if users and u.name not in users:
+                return abort(401, {'message': 'User not in allowed_users in login_required request'})
+            if roles and set(u.roles).intersection(roles):
+                return abort(401, {'message': 'User role not in allowed_roles in login_required request'})
+            return fn(*args, **kwargs)
+        return decorator
+    if f is None:
+        return login_required_inner
+    else:
+        assert callable(f)
+        login_required_inner(f)
+
+
+# Add login, logout and loggedinas routes
+blp = Blueprint("auth", __name__)
+
+
+@blp.route('/login', methods=['GET'])
+@no_cache
+def get_login_mask():
+    # return a login mask
+    return render_template_string("""
+                  <!DOCTYPE html>
+                  <html>
+                  <body>
+                    <form name="login" action="/login" method="post" accept-charset="utf-8">
+                      <div><label for="username" class="form-label">User name: </label><input type="text" name="username" placeholder="name" required="required"></div>
+                      <div><label for="password" class="form-label">Password</label><input type="password" name="password" placeholder="password" required="required"></div>
+                      <div><input type="reset" value="Reset"><input type="submit" value="Login"></div>
+                    </form>
+                  </body>
+                  </html>
+                 """)
+
+
+@blp.route('/login', methods=['POST'])
+@no_cache
+def login():
+    # Login the user with the current username/password
+    user = phoveaSecurityManager().login(request.values['username'], request.values)
+    if not user:
+        return abort(401)  # 401 Unauthorized
+
+    # Create a new user response with the name and roles
+    response = jsonify(name=user.name, roles=user.roles)
+    # Create a new access token for this user
+    access_token = create_access_token(user.name)
+    # Append the access token as cookie
+    set_access_cookies(response, access_token)
+    return response
+
+
+@login_required()
+@blp.route('/logout', methods=['GET', 'POST'])
+@no_cache
+def logout():
+    verify_jwt_in_request()
```

Comment:
As we already do this in the `current_user` property (triggered by the next call), can we remove it here?